### PR TITLE
refactor: don't throw errors from datasource package releases lookup

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Wraps getPkgReleases so that lookupUpdates should not receive Errors thrown.

## Context

First step towards separating out lookup from processing

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
@viceice should not be breaking for your internal tools as it _wraps_ `getPkgReleases` rather than refactors it?

**Comment by viceice** [comment]
no problem, I can fix it on tools 😅

File: lib/modules/datasource/types.ts [hunk 1]
```diff
@@ -74,6 +74,16 @@ export interface ReleaseResult {
   replacementVersion?: string;
 }
 
+export interface GetPkgReleasesResultError {
+  error: Error;
+  result?: undefined;
+}
+
+export interface GetPkgReleasesResultSuccess {
+  error?: undefined;
```
Comment (viceice) [inline lib/modules/datasource/types.ts:83]:
```suggestion
  error?: never;
```

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**